packages/interactivity-router/src/index.ts

```diff
@@ -295,6 +284,7 @@ export const { state, actions } = store< Store >( 'core/router', {
 				! page.initialData?.config?.[ 'core/router' ]
 					?.clientNavigationDisabled
 			) {
+				yield importScriptModules( page.scriptModules );
```

Comment:
Is this here because it is fixing a bug?

Reply:
Yes, that's fixing a bug. If we don't wait for the new modules to be ready before rendering the new page, directives like `data-wp-init` won't work, because the referenced callback doesn't exist yet during the first render.